src/libsync/syncengine.cpp

```diff
@@ -912,7 +912,16 @@ void SyncEngine::slotDiscoveryJobFinished(int discoveryResult)
             return;
         }
     }
-    if (!_hasForwardInTimeFiles && _backInTimeFiles >= 2) {
+
+    auto databaseFingerprint = _journal->dataFingerprint();
+    // If databaseFingerprint is null, this means that there was no information in the database
+    // (for example, upgrading from a previous version, or first sync)
+    // If it is empty (""), then that meanst that it was empty on the server before.
```

Comment:
Isn't empty also isNull? I'm not sure anymore since doing PHP :)
Maybe make it more clear by having an else if(isEmpty) ?